src/m365/pp/commands/aibuildermodel/aibuildermodel-list.spec.ts

```diff
@@ -0,0 +1,181 @@
+        "msdyn_aimodelid": "08ffffbe-ec1c-4e64-b64b-dd1db926c613",
+        "_msdyn_activerunconfigurationid_value": null,
+        "overriddencreatedon": null,
+        "_msdyn_retrainworkflowid_value": null,
+        "importsequencenumber": null,
+        "_msdyn_scheduleinferenceworkflowid_value": null,
+        "_modifiedonbehalfby_value": null,
+        "utcconversiontimezonecode": null,
+        "_createdonbehalfby_value": null,
+        "_owningteam_value": null,
+        "timezoneruleversionnumber": null,
+        "iscustomizable": {
+          "Value": true,
+          "CanBeChanged": true,
+          "ManagedPropertyLogicalName": "iscustomizableanddeletable"
+        }
+      }
+    ]
+  };
+  //#endregion
+
+  let log: string[];
+  let logger: Logger;
+  let loggerLogSpy: sinon.SinonSpy;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(telemetry, 'trackEvent').callsFake(() => { });
+    sinon.stub(pid, 'getProcessName').callsFake(() => '');
+    auth.service.connected = true;
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogSpy = sinon.spy(logger, 'log');
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.get,
+      powerPlatform.getDynamicsInstanceApiUrl
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      telemetry.trackEvent,
+      pid.getProcessName
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name.startsWith(commands.AIBUILDERMODEL_LIST), true);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('defines correct properties for the default output', () => {
+    assert.deepStrictEqual(command.defaultProperties(), ['msdyn_name', 'msdyn_aimodelid', 'createdon', 'modifiedon']);
+  });
+
+  it('retrieves AI Builder models', async () => {
+    sinon.stub(powerPlatform, 'getDynamicsInstanceApiUrl').callsFake(async () => envUrl);
+
+    sinon.stub(request, 'get').callsFake(async opts => {
+      if ((opts.url === `https://contoso-dev.api.crm4.dynamics.com/api/data/v9.0/msdyn_aimodels?$filter=iscustomizable/Value eq true`)) {
+        if (opts.headers &&
+          opts.headers.accept &&
+          (opts.headers.accept as string).indexOf('application/json') === 0) {
+          return modelsResponse;
+        }
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { debug: true, environment: validEnvironment } });
+    assert(loggerLogSpy.calledWith(modelsResponse.value));
+
+  });
+
+  it('retrieves AI Builder models as admin', async () => {
```

Comment:
In fact this is exactly the same test as the one above, I'd say we remove this one since it doesn't add much value.